addons/pos_self_order/models/product_product.py

```diff
@@ -19,7 +19,9 @@ class ProductTemplate(models.Model):
     @api.constrains('available_in_pos')
     def _check_combo_inclusions(self):
         super()._check_combo_inclusions()
-        self.self_order_available = False
+        for record in self:
+            if not record.available_in_pos:
+                record.self_order_available = False
```

Comment:
Why is this kind of mutation in a constraint method? Constraint should not introduce any side effect.

Do this kind of logic in a compute method. Make it a computed field and store=True.